keras/optimizers.py

```diff
@@ -69,6 +70,9 @@ def get_updates(self, params, constraints, loss):
 
     def get_gradients(self, loss, params):
         grads = K.gradients(loss, params)
+        if hasattr(self, 'normalize_gradients') and self.normalize_gradients:
+            norm = K.sqrt(sum([K.sum(K.square(g)) for g in grads])) + K.epsilon()
```

Comment:
Btw, here is another use-case: https://arxiv.org/pdf/1707.04199.pdf. Here, the gradients are exponentiated while keeping output activation linear instead of softmax. With `gradient_modfifier_fn` this is easy to handle as well.